arangod/Pregel/MetricsActor.h

```diff
@@ -0,0 +1,230 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2014-2023 ArangoDB GmbH, Cologne, Germany
+/// Copyright 2004-2014 triAGENS GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Aditya Mukhopadhyay
+////////////////////////////////////////////////////////////////////////////////
+
+#pragma once
+
+#include "Actor/HandlerBase.h"
+#include "Pregel/MetricsMessages.h"
+#include "Pregel/PregelMetrics.h"
+#include "Logger/LogMacros.h"
+
+namespace arangodb::pregel {
+struct MetricsState {
+  std::shared_ptr<PregelMetrics> metrics;
+};
+
+template<typename Runtime>
+struct MetricsHandler : actor::HandlerBase<Runtime, MetricsState> {
+  auto operator()([[maybe_unused]] message::MetricsStart msg) {
+    LOG_TOPIC("89eac", INFO, Logger::PREGEL)
+        << fmt::format("Metric Actor {} started", this->self);
+
+    return std::move(this->state);
+  }
+
+  template<message::Gauge G, message::GaugeOp O>
+  auto operator()(message::GaugeUpdate<G, O> gauge)
```

Comment:
I think it would be more readable to have more specific messages instead of the generic GaugeUpdate ones, e.g. LoadingStarted, StoringStarted and so on. Then you don't need the visit and e.g. for StoringStarted you can increase the storing counter and decrease the computing counter.

Reply:
Done.